src/Nethermind/Nethermind.Network/ForkInfo.cs

```diff
@@ -3,36 +3,51 @@
 
 using System;
 using System.Buffers.Binary;
+using System.Collections.Generic;
 using System.Linq;
 using System.Numerics;
 using System.Runtime.CompilerServices;
 using Force.Crc32;
+using Nethermind.Blockchain;
+using Nethermind.Core;
 using Nethermind.Core.Collections;
 using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
 using Nethermind.Specs;
 
 namespace Nethermind.Network
 {
-    public class ForkInfo
+    public class ForkInfo : IForkInfo
     {
+        private Dictionary<uint, (ForkActivation Activation, ForkId Id)> DictForks { get; }
         private (ForkActivation Activation, ForkId Id)[] Forks { get; }
 
-        public ForkInfo(ISpecProvider specProvider, Keccak genesisHash)
+        private IBlockTree _blockTree;
+        private ulong TimstampThreshold => 1438269973ul; // MainnetSpecProvider.ShanghaiBlockTimestamp;
```

Comment:
We can't just bake it here! What about other networks like Gnosis?

Reply:
Why not, it is not like there any other networks that have timestamp forks before this